redbot/cogs/audio/audio.py

```diff
@@ -2671,6 +2746,33 @@ def _track_creator(player, position=None, other_track=None):
             track_obj[key] = value
         return track_obj
 
+    @staticmethod
+    def _track_limit(ctx, track, maxlength):
+        try:
+            length = round(track.length / 1000)
+        except AttributeError:
+            length = round(track / 1000)
+        if length > 259200:
+            return True
+        elif length >= maxlength:
+            return False
+        else:
+            return True
+
+    async def _time_convert(self, length):
+        match = re.compile(r"(?:(\d+):)?([0-5][0-9]):([0-5][0-9])").match(length)
```

Comment:
Replacing this regex with `(?:(\d+):)?([0-5]?[0-9]):([0-5][0-9])` can help improve usability where users can supply `8:00` instead of `08:00` for setting the max length a minor usability change.

Reply:
Thank you for this.